src/bin/pg_autoctl/keeper.c

```diff
@@ -477,17 +535,25 @@ keeper_update_pg_state(Keeper *keeper)
 
 	/*
 	 * In some states, PostgreSQL isn't expected to be running, or not expected
-	 * to have a streaming replication to monitor at all.
+	 * to have a streaming replication to monitor at all. When the streaming
+	 * replication is running as expected then postgres->pgsrSyncState is
+	 * filled in, otherwise it's an empty string.
 	 */
 	switch (keeperState->current_role)
 	{
 		case PRIMARY_STATE:
-		case WAIT_PRIMARY_STATE:
+		{
```

Comment:
wonder if this change has anything to do with the behavior we are observing

Reply:
We don't use the return value anyway, if I'm reading the code correctly. The update failure would not be fatal here. Not sure about this having such an impact.